samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -311,6 +321,68 @@ private StreamPartitionCountMonitor getPartitionCountMonitor(Config config, Syst
       });
   }
 
+  private Optional<InputStreamRegexMonitor> getInputRegexMonitor(Config config, SystemAdmins systemAdmins) {
+
+    // if input regex monitor is not enabled return empty
+    if (new JobConfig(config).getMonitorInputRegexFrequency() <= 0) {
+      log.info("InputStreamRegexMonitor is disabled.");
+      return Optional.empty();
+    }
+
+    StreamMetadataCache streamMetadata = new StreamMetadataCache(systemAdmins, 0, SystemClock.instance());
+    Set<SystemStream> inputStreamsToMonitor = new TaskConfigJava(config).getAllInputStreams();
+    if (inputStreamsToMonitor.isEmpty()) {
+      throw new SamzaException("Input streams to a job can not be empty.");
+    }
+
+    // Compile a map of each input-system to its corresponding input-monitor-regex patterns
+    Map<String, Pattern> inputRegexesToMonitor = new HashMap<>();
+
+    // First list all rewriters
+    Option<String> rewritersList = new JobConfig(config).getConfigRewriters();
+
+    // if no rewriter is defined, there is nothing to monitor
+    if (rewritersList.isDefined()) {
+
+      String[] rewriters = rewritersList.get().split(",");
```

Comment:
encapsulate this in a method inside JobConfig

Map<String, Pattern> getRegexSystemStreams();

Reply:
done